### PR TITLE
Normative: Perform legacy hoisting for `function arguments() {}` 

### LINKED ISSUE #815: Annex B.3.3.1 intention with `arguments` parameter shadowing
(Credit to @anba for discovering this in [Firefox's bug 1339123](https://bugzilla.mozilla.org/show_bug.cgi?id=1339123))

Consider the following two programs:

```javascript
(function (a = arguments) {
  print(arguments);
  var arguments;
}());
```

and

```javascript
(function (a = arguments) {
  print(arguments);
  { function arguments() { } }
}());
```

According to the current spec text (ES2017), both programs should print `[object Arguments]`.
According to ES2015, the second program prints `undefined`.

PR #677 from @jswalden resulted in the semantics change, though he confirmed over IRC any semantics changes from the PR were unintended.

What is the intended behavior here? I am of the opinion both programs should print `[object Arguments]`. That is, that the current spec is correct, and ES2015 had an inconsistency bug that has since been fixed.

For completeness, SpiderMonkey prints `[object Arguments]` for both. V8 prints `undefined` for both. I don't have JSC and Chakra handy, but @anba says they both print `[object Arguments]` for both tests.

### PR DESCRIPTION
**Pull request by littledan**
An editorial change accidentally blocked this hoisting; this patch
restores it. Corresponding tests already pass on JSC and ChakraCore.

Closes #815

### CODE REVIEW

**Comment by littledan** [comment]
Reopening this PR with a pretty ad-hoc solution to the issue raised on the previous thread #889: Annex B 3.3.1 is augmented with skipping setting arguments to undefined. It's unfortunate to have a special case here, but arguments is already so special in FunctionDeclarationInstantiation...

File: spec.html [hunk 1]
```diff
@@ -12926,6 +12928,7 @@
         <li>If _base_ is *-0* and _exponent_ &lt; 0 and _exponent_ is an odd integer, the result is *-&infin;*.</li>
         <li>If _base_ is *-0* and _exponent_ &lt; 0 and _exponent_ is not an odd integer, the result is *+&infin;*.</li>
         <li>If _base_ &lt; 0 and _base_ is finite and _exponent_ is finite and _exponent_ is not an integer, the result is *NaN*.</li>
+        <li> In the remaining cases, the result is an approximation of raising _base_ to the power _exponent_, where the difference between the result and the mathematical value is less than 1 ulp.</li>
```
Comment (bterlson) [inline spec.html]: This seems like an unrelated change.

Comment (littledan) [inline spec.html]: Sorry for that; thanks for landing this properly.
